scripts/docker_build/sof_builder/Dockerfile

```diff
@@ -121,9 +120,19 @@ RUN cd "$HOME" && \
 		make && make install && \
 		rm -rf etc/config.cache; \
 	done && \
-	cd "$HOME" && rm -rf newlib-xtensa && \
 	echo "Stage3: newlib-xtensa is done!"
 
+# log all hashes for each repo to a file
+ARG SOF_WORK="$HOME"/work
+ARG HASH_LIST="$SOF_WORK"/sof_docker_hash.txt
+RUN echo "Build date: $(date +%Y%m%d)" > "$HASH_LIST" && \
+	cd "$SOF_WORK"/alsa/alsa-lib && pwd >> "$HASH_LIST"; git log --oneline -5 >> "$HASH_LIST" && \
+	cd "$SOF_WORK"/alsa/alsa-utils && pwd >> "$HASH_LIST"; git log --oneline -5 >> "$HASH_LIST" && \
+	cd "$HOME"/xtensa-overlay && pwd >> "$HASH_LIST"; git log --oneline -5 >> "$HASH_LIST" && \
+	cd "$HOME"/crosstool-ng && pwd >> "$HASH_LIST"; git log --oneline -5 >> "$HASH_LIST" && \
+	cd "$HOME"/newlib-xtensa && pwd >> "$HASH_LIST"; git log --oneline -5 >> "$HASH_LIST" && \
+	cd "$HOME" && rm -rf xtensa-overlay crosstool-ng newlib-xtensa && \
+	echo "Stage4: hash logging and deleting repos are done!"
```

Comment:
I think you can make this a `for` loop:

```shell
set -e ; for i in "$SOF_WORK"/alsa/alsa-lib "$SOF_WORK"/alsa/alsa-utils ...; do cd "$i"; git log ... ; done >> "$HASH_LIST"
```

Reply:
I tried to keep "" around the variables but it is not expanding if I do so. shellcheck warning is valid but we know $HOME and $SOF_WORK do not have space in it. I think better than all hard-coded path.

```shell
ARG REPOS="/home/sof/work/alsa/alsa-lib /home/sof/work/alsa/alsa-utils /home/sof/xtensa-overlay /home/sof/crosstool-ng /home/sof/newlib-xtensa"
```